src/_components/ColorBar.html.svelte

```diff
@@ -0,0 +1,191 @@
+<!--
+	@component
+	Creates a color bar key for chromatic scales on `zScale`.
+ -->
+<script>
+	import { getContext } from 'svelte';
+
+	const { zScale } = getContext('LayerCake');
+
+	/** @type {string | null} [label=null] - Text label to show next to the color bar */
+	export let label = null;
+
+	/** @type {string} [labelSide='left'] - Position of the label. Can be 'top', 'right', 'bottom', or 'left
+	 * text label relative to the gradient bar */
+	export let labelSide = `left`;
+
+	/** @type {Number|Array|Function} [ticks=4] - If this is a number, it passes that along to the [d3Scale.ticks](https://github.com/d3/d3-scale) function. If this is an array, hardcodes the ticks to those values. If it's a function, passes along the default tick values and expects an array of tick values in return. */
+	export let ticks = 3;
+
+	/** @type {Boolean} [tickMarks=false] - Show a vertical mark for each tick. */
+	export let tickMarks = false;
+
+	/** @type {Function} [formatTick=d => d] - A function that passes the current tick value and expects a nicely formatted value in return. */
+	export let formatTick = (d) => d;
+
+	/** @type {Boolean} [snapTicks=true] - Instead of centering the text on the first and the last items, align them to the edges of the chart. */
+	export let snapTicks = true;
+
+	/** @type {string} [tickSide='bottom'] - Position of tick labels. Can be 'top' or 'bottom' */
+	export let tickSide = `bottom`;
+
+	/** @type {Number} [steps=100] - Number of samples to take of the color ramp to create the linear gradient */
+	export let steps = 100;
+
+	$: tickVals = Array.isArray(ticks)
+		? ticks
+		: typeof ticks === 'function'
+		? ticks($zScale.ticks())
+		: $zScale.ticks(ticks);
+
+	$: ramped = [...Array(steps).keys()].map((i) => $zScale(i / steps));
```

Comment:
@janosh I think the gradient creation needs some tweaks. I was testing it out with `scaleDiverging` using a domain of `[-1, 0, 1]` and because the gradient uses `i` here, it will always start at `0` and never get negative values. It should instead use the tick values. This should be a good example to work from: https://observablehq.com/@d3/color-legend